lib/app/data/repository/auth/auth_repository_impl.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:asset_tracker/app/core/utils/constants/app_texts.dart';
+import 'package:asset_tracker/app/data/models/auth/user_model.dart';
+import 'package:asset_tracker/app/data/repository/auth/auth_repository.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:asset_tracker/app/core/exception/firebaseauth_exception.dart';
+
+class AuthRepositoryImpl implements AuthRepository {
+  final FirebaseAuth _firebaseAuth = FirebaseAuth.instance;
+
+  @override
+  Future<UserModel> signUp(String email, String password) async {
+    try {
+      final result = await _firebaseAuth.createUserWithEmailAndPassword(
+        email: email,
+        password: password,
+      );
+      return UserModel(result.user?.email ?? "", password);
+    } on FirebaseAuthException catch (e) {
+      if (e.code == AppTexts.userNotFound) {
```

Comment:
final isUserNotFound = [AppTexts.userNotFound,AppTexts.wrongPassword].contains(e.code);
if(isUserNotFound)....

Reply:
not hem if içerisinde hem else if içerisinde aynı mesajı dönmüşsün. Eğer kasıtlı yaptıysan problem yok ama copy paste yaparken değiştirmeyi unuttuysan aklında bulunsun